core/src/main/java/org/mule/runtime/core/api/processor/MessageProcessors.java

```diff
@@ -41,64 +44,68 @@ private MessageProcessors() {
-   * construction but wil not inject {@link MuleContext} or {@link FlowConstruct} or perform any lifecycle.
+   * construction but will not inject {@link MuleContext} or perform any lifecycle.
    *
+   * @param processingStrategy the processing strategy to use for configuring the chain. It may be {@link Optional#empty()}.
    * @param processors list of processors to construct chains from.
    * @return new {@link MessageProcessorChain} instance.
    */
-  public static MessageProcessorChain newChain(List<Processor> processors) {
-    if (processors.size() == 1 && processors.get(0) instanceof MessageProcessorChain
-        && !(processors.get(0) instanceof ExplicitMessageProcessorChain)) {
-      return (MessageProcessorChain) processors.get(0);
-    } else {
-      return new DefaultMessageProcessorChainBuilder().chain(processors).build();
-    }
+  public static MessageProcessorChain newChain(Optional<ProcessingStrategy> processingStrategy, Processor... processors) {
+    return newChain(processingStrategy, asList(processors));
   }
 
   /**
    * Creates a new explicit {@link MessageProcessorChain} from one or more {@link Processor}'s. Note that this performs chains
-   * construction but wil not inject {@link MuleContext} or {@link FlowConstruct} or perform any lifecycle.
+   * construction but will not inject {@link MuleContext} or perform any lifecycle.
    *
    * @param processors list of processors to construct chains from.
    * @return new {@link MessageProcessorChain} instance.
    */
-  public static MessageProcessorChain newExplicitChain(Processor... processors) {
+  public static MessageProcessorChain newExplicitChain(Optional<ProcessingStrategy> processingStrategy, Processor... processors) {
     if (processors.length == 1 && processors[0] instanceof ExplicitMessageProcessorChain) {
       return (MessageProcessorChain) processors[0];
     } else {
-      return new ExplicitMessageProcessorChainBuilder().chain(processors).build();
+      ExplicitMessageProcessorChainBuilder explicitMessageProcessorChainBuilder = new ExplicitMessageProcessorChainBuilder();
+      processingStrategy.ifPresent(explicitMessageProcessorChainBuilder::setProcessingStrategy);
+      return explicitMessageProcessorChainBuilder.chain(processors).build();
     }
   }
 
   /**
    * Creates a new explicit {@link MessageProcessorChain} from a {@link List} of {@link Processor}'s. Note that this performs
-   * chains construction but wil not inject {@link MuleContext} or {@link FlowConstruct} or perform any lifecycle.
+   * chains construction but wil not inject {@link MuleContext} or perform any lifecycle.
    *
    * @param processors list of processors to construct chains from.
    * @return new {@link MessageProcessorChain} instance.
    */
-  public static MessageProcessorChain newExplicitChain(List<Processor> processors) {
+  public static MessageProcessorChain newExplicitChain(Optional<ProcessingStrategy> processingStrategy,
```

Comment:
Q: Do we still 'explicit chains'? This was added to be able to have different flow paths/component location depending on is chain was internal or created by user. Unsure if this is still needed though.